content/how-to/hot-module-reload.md

```diff
@@ -0,0 +1,186 @@
+    "babel-loader": "^6.2.4",
+    "babel-preset-es2015-webpack": "^6.4.2",
+    "babel-preset-react": "^6.11.1",
+    "babel-preset-stage-2": "^6.11.0",
+    "css-loader": "^0.23.1",
+    "postcss-loader": "^0.9.1",
+    "react-hot-loader": "^3.0.0-beta.1",
+    "style-loader": "^0.13.1",
+    "webpack": "^2.1.0-beta.20",
+    "webpack-dev-server": "^2.1.0-beta.0"
+  }
+```
+
+###Babel Config
+Your `.babelrc` file should look like the following:
+
+```js
+{
+  "presets": ["es2015-webpack", "stage-2", "react"],
+  "plugins": ["react-hot-loader/babel"]
+}
+```
+
+There's a few things to note here. First, in the presets, we're using es2015-webpack, which is a Babel preset that enables support for es2015 while DISABLING conversion to CommonJS module format. Since Webpack 2 understands import/export syntax, the es2015-webpack preset (unlike the general es2015 preset) allows Webpack to perform tree-shaking and dead code elimination.
+
+The stage-2 preset is up to your comfort with adopting new technoligies; stage-0 will keep you on the bleeding age.
+
+Finally, the React preset will enable React specific transpilation, while the react-hot-loader/babel plugin will enable your React code to work with HMR.
+
+###Webpack config
+While there's many ways of setting up your Webpack config - via API, via multiple or single config files, etc - here is the basic information you should have available:
+
+```js
+const { resolve } = require('path');
+const webpack = require('webpack');
+
+const ENTRY = ['./index.js'];
+
+module.exports = env => {
+  return {
+    entry: env.dev?
+      [
+        'react-hot-loader/patch',
+        'webpack-dev-server/client?http://localhost:8080',
+        'webpack/hot/only-dev-server',
+      ].concat(ENTRY)
+      :
+      ENTRY,
+    output: {
+      filename: 'bundle.js',
+      path: resolve(__dirname, 'dist'),
+      publicPath: '/'
+    },
+    context: resolve(__dirname, 'src'),
+    devtool: env.prod ? 'source-map' : 'inline-source-map',
+    devServer: {
+      hot: true,
+      publicPath: '/'
+    },
+    module: {
+      loaders: [
+        { test: /\.js$/, loader: 'babel',
+          exclude: /node_modules/ },
+        {
+          test: /\.css$/,
+          loader: 'style!css-loader?modules&postcss-loader',
+        },
+      ],
+    },
+    plugins: [
+      new webpack.HotModuleReplacementPlugin(),
+      new webpack.NamedModulesPlugin(),
+    ],
+  }
+};
+```
+
+There's a lot going on above, and not all of it is related to HMR. If you are new to Webpack configs, please see the documentation elsewhere on this site. The basic assumption here is that your JS entry is located at `./src/index.js`, and that you're using CSS Modules for your styling. Let's look at the HMR specific items.
+
+First, the plugins. While only the HMRPlugin itself is necessary to enable HMR, the NamedModulesPlugin will make your life easier by showing an intelligible module id in your browser console whenever a file is changed.
```

Comment:
Expand `HMRPlugin` to full name. That `NamedModulesPlugin` bit can be eliminated by moving the explanation as a code comment.